atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/SingleBusyCellWorkflows.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.workload.workflow;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.workload.store.ImmutableWorkloadCell;
+import com.palantir.atlasdb.workload.store.InteractiveTransactionStore;
+import com.palantir.atlasdb.workload.store.ReadOnlyTransactionStore;
+import com.palantir.atlasdb.workload.store.WorkloadCell;
+import com.palantir.atlasdb.workload.transaction.DeleteTransactionAction;
+import com.palantir.atlasdb.workload.transaction.WriteTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransaction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransactions;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This workflow attempts to manipulate a single cell in a table, scheduling transactions that perform writes and
+ * deletes, along with transactions that simply read. Notice that unlike other workflows, because the load is
+ * considerably heterogeneous, we do not use a {@link DefaultWorkflow} to schedule transactions.
```

Comment:
Do you think we should make a new type of abstract workflow to support this? I don't feel strongly either way, and think this is fine as it is.

Reply:
Alternatively, you could use `DefaultWorfklow` and just do `index % 4` for writes, and `index % 2` for reads?

Reply:
Ugh I forgot that `KeyedTransactionTask` only returns a single transaction :/

Reply:
So the tricky bit here is that I actually wanted separate executors for the reads and writes, so that we have both types of transactions running in parallel. (One could say maybe we can rely on the Antithesis environment to find the interesting bugs, though given we might be also using this outside of antithesis I think it's good to have places in our code that push away from less interesting cases continue to do so!)

Reply:
We could make a new type of abstract workflow that supports different types of workloads and has an executor for each type, though I'd wait for us to have one or two more that need it

Reply:
OK yup that's fine